src/main/java/org/cloudfoundry/promregator/lite/cfaccessor/SimpleCFAccessorImpl.kt

```diff
@@ -0,0 +1,511 @@
+                val getInfo = client.info()[request].block()
+                        ?: throw RuntimeException("Error connecting to CF api '$api'")
+                // NB: This also ensures that the connection has been established properly...
+                log.info {"Target CF platform ($api) is running on API version ${getInfo.apiVersion}"}
+
+                // Ensures v3 API exists. The CF Java Client does not yet implement the info endpoint for V3, so we do it manually.
+                val v3Info = ReactorInfoV3(client.connectionContext, client.rootV3,
+                        client.tokenProvider, client.requestTags)
+                        .get().onErrorReturn(JsonNodeFactory.instance.nullNode()).block()
+                if (v3Info == null || v3Info.isNull) {
+                    log.warn("Unable to get v3 info endpoint of CF platform, some features will not work as expected")
+                    _isV3Enabled = false
+                } else {
+                    _isV3Enabled = true
+                }
+            }
+        }
+    }
+
+    private fun connectionContext(apiConfig: ApiConfig, proxyConfiguration: ProxyConfiguration?): DefaultConnectionContext {
+        if (PATTERN_HTTP_BASED_PROTOCOL_PREFIX.matcher(apiConfig.host).find()) {
+            throw ConfigurationException("cf.api_host configuration parameter must not contain an http(s)://-like prefix; specify the hostname only instead")
+        }
+        var connctx = DefaultConnectionContext.builder()
+                .apiHost(apiConfig.host)
+                .skipSslValidation(apiConfig.skipSslValidation)
+        if (proxyConfiguration != null) {
+            connctx = connctx.proxyConfiguration(proxyConfiguration)
+        }
+        if (apiConfig.connectionPool.size != null) {
+            connctx = connctx.connectionPoolSize(apiConfig.connectionPool.size)
+        }
+        if (apiConfig.threadPool.size != null) {
+            connctx = connctx.threadPoolSize(apiConfig.threadPool.size)
+        }
+        return connctx.build()
+    }
+
+    private fun tokenProvider(apiConfig: ApiConfig): PasswordGrantTokenProvider {
+        return PasswordGrantTokenProvider.builder().password(apiConfig.password).username(apiConfig.username).build()
+    }
+
+    private fun proxyConfiguration(apiConfig: ApiConfig): ProxyConfiguration? {
+        val effectiveProxyHost = apiConfig.proxy?.host
+        val effectiveProxyPort = apiConfig.proxy?.port ?: 0
+
+        if (effectiveProxyHost != null && PATTERN_HTTP_BASED_PROTOCOL_PREFIX.matcher(effectiveProxyHost).find()) {
+            throw ConfigurationException("Configuring of cf.proxyHost or cf.proxy.host configuration parameter must not contain an http(s)://-like prefix; specify the hostname only instead")
+        }
+        return if (effectiveProxyHost != null && effectiveProxyPort != 0) {
+            var proxyIP: String? = null
+            // TODO add back InetAddressUtils
```

Comment:
I removed the InetAddressUtils since it had been the only usage of the apache lib in the lite project. I'll add it back.